src/main/java/nextstep/subway/section/domain/SortedSections.java

```diff
@@ -0,0 +1,66 @@
+package nextstep.subway.section.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public class SortedSections {
+    private List<Section> sections;
+
+    public SortedSections(List<Section> sections) {
+        this.sections = new ArrayList<>(sections);
+        sort();
+    }
+
+    private void sort() {
+        if (sections.isEmpty()) {
+            return;
+        }
+
+        Section topSection = getTopSection(sections.get(0));
+
+        this.sections = sort(topSection);
+    }
+
+    private Section getTopSection(Section firstSection) {
+        for (Section section : sections) {
+            if (firstSection == section) continue;
+
+            if (firstSection.isLower(section)) {
+                return getTopSection(section);
+            }
+        }
+
+        return firstSection;
+    }
+
+    private List<Section> sort(Section upperSection) {
+        List<Section> sortedList = new ArrayList<>();
+        sortedList.add(upperSection);
+
+        Section bottomSection = findBottomSection(upperSection);
+
+        if (bottomSection == null) {
+            return sortedList;
+        }
+
+        sortedList.addAll(sort(bottomSection));
+        return sortedList;
+    }
+
+    private Section findBottomSection(Section of) {
+        for (Section section : sections) {
```

Comment:
이 부분도 람다식을 통해 개선해 볼 수 있을 것 같습니다 😄 
```suggestion
        return sections.stream()
                .filter(of::isUpper)
                .findFirst()
                .orElse(null);
```